satpy/tests/writer_tests/test_cf.py

```diff
@@ -829,18 +831,75 @@ def _gm_matches(gmapping, expected):
             area_id='geos',
             description='geos',
             proj_id='geos',
-            projection={'proj': 'geos', 'h': h, 'datum': 'WGS84', 'ellps': 'GRS80'},
+            projection={'proj': 'geos', 'h': h, 'datum': 'WGS84', 'ellps': 'GRS80',
+                        'lat_0': 0, 'lon_0': 0},
+            width=2, height=2,
+            area_extent=[-1, -1, 1, 1])
+        geos_expected = xr.DataArray(data=0,
+                                     attrs={'perspective_point_height': h,
+                                            'latitude_of_projection_origin': 0,
+                                            'longitude_of_projection_origin': 0,
+                                            'grid_mapping_name': 'geostationary',
+                                            # 'semi_major_axis': 6378137.0,
+                                            # 'semi_minor_axis': 6356752.314,
+                                            # 'sweep_angle_axis': None,
+                                            })
+
+        ds = ds_base.copy()
+        ds.attrs['area'] = geos
+        res, grid_mapping = area2gridmapping(ds)
+
+        self.assertEqual(res.attrs['grid_mapping'], 'geos')
+        _gm_matches(grid_mapping, geos_expected)
+
+        # e) oblique Mercator
+        area = pyresample.geometry.AreaDefinition(
+            area_id='omerc_otf',
+            description='On-the-fly omerc area',
+            proj_id='omerc',
+            projection={'alpha': '9.02638777018478', 'ellps': 'WGS84', 'gamma': '0', 'k': '1',
+                        'lat_0': '-0.256794486098476', 'lonc': '13.7888658224205',
+                        'proj': 'omerc', 'units': 'm'},
+            width=2837,
+            height=5940,
+            area_extent=[-1460463.0893, 3455291.3877, 1538407.1158, 9615788.8787]
+        )
+
+        omerc_dict = {'azimuth_of_central_line': 9.02638777018478,
+                      'false_easting': 0.,
+                      'false_northing': 0.,
+                      # 'gamma': 0,  # this is not CF compliant
+                      'grid_mapping_name': "oblique_mercator",
+                      'latitude_of_projection_origin': -0.256794486098476,
+                      'longitude_of_projection_origin': 13.7888658224205,
+                      # 'prime_meridian_name': "Greenwich",
+                      'reference_ellipsoid_name': "WGS 84"}
+        omerc_expected = xr.DataArray(data=0, attrs=omerc_dict)
+
+        ds = ds_base.copy()
+        ds.attrs['area'] = area
+        res, grid_mapping = area2gridmapping(ds)
+
+        self.assertEqual(res.attrs['grid_mapping'], 'omerc_otf')
+        _gm_matches(grid_mapping, omerc_expected)
+
+        # d) Projection that has a representation but no explicit a/b
```

Comment:
d) is after e) ...

Reply:
Turns out there was a redundant omerc test which I think came from merging/rebasing with the other PR that dealt with this functionality (not going to look it up right now). There were two c's or d's I think too. Fixed now.